packages/components/box/index.js

```diff
@@ -0,0 +1,29 @@
+/* eslint-disable indent */
+import React from 'react';
+import {
+  PseudoBox as ChakraPseudoBox,
+  ThemeProvider,
+  CSSReset,
+} from '@chakra-ui/core';
+import PropTypes from 'prop-types';
+
+const Box = ({ children, ...props }) => (
+  <ThemeProvider>
+    <CSSReset />
+    <ChakraPseudoBox {...props}>
+      {typeof children === 'function' ? children() : children}
+    </ChakraPseudoBox>
```

Comment:
```suggestion
    <ChakraPseudoBox {...props}>
      {children}
    </ChakraPseudoBox>
```
Why is that Children is written like this? for my learning.
Is this necessary?

Reply:
children could be a function that returns some component or dom elements or it could be a direct component or dom elements .

Reply:
Okay, but why need to differentiate them, 
```suggestion
    <ChakraPseudoBox {...props}>
      {children}
    </ChakraPseudoBox>
```
This is self sufficient right. @prasanna1211 your thoughts.
@govindsingh55 Any realtime code reference where we would this be useful.

Reply:
the best example could be sub-render functions

const subR = () => (<h1>hello</h1>);

const hello = () => {
 return(
  <div>{hello}</div>
)
}

Reply:
React.children can only be an array (if not object when the child is single element I suppose) https://reactjs.org/docs/react-api.html#reactchildren ??

Reply:
Sub render functions are ASTs that treat its node type as a DOM element. So it cannot be a function I believe.

Reply:
Okay, Okay, I understood but in the background react will take the both and do the rendering correctly right. even if we did not give the above line? or it wont?

Reply:
> React.children can only be an array (if not object when the child is single element I suppose) https://reactjs.org/docs/react-api.html#reactchildren ??

I will look at it.

Reply:
It will work even if we don't check it. Basically children cannot be a function. It will always be an array. (As per my understanding).

Reply:
Actually I am also not that much clear and I took it chakra GitHub repo.

Reply:
Seems React Children is only an array. No need of the above classification
All I see where it has been used is in Class components.

Reply:
> Actually I am also not that much clear and I took it chakra GitHub repo.

Can you link the source. I couldn't find it in Box & Flex component of Chakra

Reply:
I think here it's not required. 
[reference](https://github.com/chakra-ui/chakra-ui/blob/master/packages/chakra-ui/src/Menu/index.js)

Reply:
Okay. Thanks for the reference.